python3/ltk/actions/push_action.py

```diff
@@ -73,22 +73,39 @@ def _update_current_docs(self, **kwargs):
                     print('Update {0}'.format(display_name))
                     continue
                 if self.metadata_only or not self.doc_manager.is_doc_modified(entry['file_name'], self.path):
-                    response = self.api.document_update(entry['id'], doc_metadata=self.metadata, **kwargs)
+                    response, previous_doc_id = self.update_doc_response_manager(self.api.document_update(entry['id'], doc_metadata=self.metadata, **kwargs), entry['id'], doc_metadata=self.metadata, **kwargs)
                 else:
-                    response = self.api.document_update(entry['id'], os.path.join(self.path, entry['file_name']), doc_metadata=self.metadata, **kwargs)
+                    response, previous_doc_id = self.update_doc_response_manager(self.api.document_update(entry['id'], os.path.join(self.path, entry['file_name']), doc_metadata=self.metadata, **kwargs), entry['id'], os.path.join(self.path, entry['file_name']), doc_metadata=self.metadata, **kwargs)
                 if response.status_code == 202:
-                    updated += 1
-                    logger.info('Updated {0}'.format(display_name))
-                    self._update_document(entry['file_name'])
+                    try:
+                        next_document_id = response.json()['next_document_id']
+                    except Exception:
+                        next_document_id = None
+                    finally:
+                        updated += 1
+                        logger.info('Updated {0}'.format(display_name))
+                        self._update_document(entry['file_name'], next_document_id)
+                elif response.status_code == 410:
+                    self.doc_manager.remove_element(previous_doc_id)
+                    print("Document was uploaded, but ID has been archived. Reuploading")
+                    self.add.add_document(entry['file_name'], entry['name'])
                 else:
+                    failed += 1
                     raise_error(response.json(), "Failed to update document {0}".format(entry['name']), True)
 
-        return updated
+        return updated, failed
+
+    def update_doc_response_manager(self, response, document_id, *args, **kwargs):
```

Comment:
This seems to be the same function as the one in actions.py, and push_actions inherits from actions so you can probably just use the parent function instead of overriding it